Tella/Scenes/Uwazi/Utils/UwaziEntityParser.swift

```diff
@@ -8,77 +8,241 @@
+                                              required: $0.propertyRequired,
+                                              helpText: $0.translatedLabel,
+                                              selectValues: $0.values,
+                                              name: $0.name)
+            }
+            
+            entryPrompts.append(prompt)
+        }
+    }
+    
+    func saveAnswersToEntityInstance(status:EntityStatus) {
+        
+        var metadata: [String: Any] = [:]
+        var uwaziEntityInstanceFile : [UwaziEntityInstanceFile] = []
+        let entityInstance = self.entityInstance == nil ? UwaziEntityInstance() : self.entityInstance
+        
+        for entryPrompt in entryPrompts {
+            
+            guard let propertyName = entryPrompt.name else { break }
+            
+            switch entryPrompt.type {
+                
+            case .dataTypeText where propertyName == UwaziEntityMetadataKeys.title:
+                guard let entryPrompt = entryPrompt as? UwaziTextEntryPrompt else { continue }
+                entityInstance?.title = entryPrompt.value
+                
+            case .dataTypeText, .dataTypeNumeric, .dataTypeMarkdown,.dataTypeDate:
+                guard let entryPrompt = entryPrompt as? UwaziTextEntryPrompt else { continue }
+                guard !entryPrompt.isEmpty else { continue }
+                metadata[propertyName] = [UwaziValue(value: entryPrompt.value)].arraydDictionnary
+                
+            case .dataTypeSelect, .dataTypeMultiSelect:
+                guard let entryPrompt = entryPrompt as? UwaziSelectEntryPrompt else { continue }
+                guard !entryPrompt.isEmpty else { continue }
+                metadata[propertyName] = entryPrompt.value.compactMap({UwaziValue(value: $0)}).arraydDictionnary
+                
+            case .dataTypeMultiFiles:
+                guard let entryPrompt = entryPrompt as? UwaziFilesEntryPrompt else { continue }
+                let attachments = entryPrompt.value
+                let attachedVaultFiles = attachments.compactMap({UwaziEntityInstanceFile(vaultFileInstanceId: $0.id , entityInstanceId:self.entityInstance?.id )})
+                uwaziEntityInstanceFile.append(contentsOf: attachedVaultFiles)
+                entityInstance?.attachments = attachments
+            case .dataTypeMultiPDFFiles:
+                guard let entryPrompt = entryPrompt as? UwaziFilesEntryPrompt else { continue }
+                let attachments = entryPrompt.value
+                let attachedVaultFiles = attachments.compactMap({UwaziEntityInstanceFile(vaultFileInstanceId: $0.id , entityInstanceId:self.entityInstance?.id )})
+                uwaziEntityInstanceFile.append(contentsOf: attachedVaultFiles)
+                entityInstance?.attachments = attachments
+            case .dataRelationship:
+                guard let entryPrompt = entryPrompt as? UwaziRelationshipEntryPrompt else { continue }
```

Comment:
Could you use the `UwaziValue` 

`metadata[propertyName] = entryPrompt.value.compactMap({UwaziValue(value: $0)}).arraydDictionnary
`